Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/ParameterWindow.cpp

```diff
@@ -917,20 +918,25 @@ namespace EMStudio
 
         const AZStd::string oldName = parameter->GetName();
 
-        // create and init the dialog
-        ParameterCreateEditDialog* dialog = new ParameterCreateEditDialog(m_plugin, this, parameter);
-        dialog->Init();
-        // We cannot use exec here as we need to access it from the tests
-        EMStudio::ParameterCreateEditDialog::connect(dialog, &QDialog::finished, [=](int resultCode)
+        // create and init the widget
+        if (!m_parameterCreateEditWidget)
```

Comment:
yeah. Same comment as above. Already we saw the similar code pop up multiple times. I would strongly consider just create the ParameterCreateEditWidget on initialization, and ReInit it with different parameter each time.